internal/entity/article.go

```diff
@@ -0,0 +1,122 @@
+package entity
+
+import (
+	"encoding/xml"
+)
+
+type (
+	ArticleListExternal struct {
+		XMLName        xml.Name                    `xml:"NewListInformation"`
+		ClubName       string                      `xml:"ClubName"`
+		ClubWebsiteURL string                      `xml:"ClubWebsiteURL"`
+		ArticleList    NewsletterNewsItemsExternal `xml:"NewsletterNewsItems"`
+	}
+
+	NewsletterNewsItemsExternal struct {
+		XMLName  xml.Name                      `xml:"NewsletterNewsItems"`
+		Articles []*NewsletterNewsItemExternal `xml:"NewsletterNewsItem"`
+	}
+
+	NewsletterNewsItemExternal struct {
+		XMLName           xml.Name `xml:"NewsletterNewsItem"`
+		ArticleURL        string   `xml:"ArticleURL"`
+		NewsArticleID     int64    `xml:"NewsArticleID"`
+		PublishDate       string   `xml:"PublishDate"`
+		Taxonomies        string   `xml:"Taxonomies"`
+		TeaserText        string   `xml:"TeaserText"`
+		ThumbnailImageURL string   `xml:"ThumbnailImageURL"`
+		Title             string   `xml:"Title"`
+		OptaMatchId       string   `xml:"OptaMatchId"`
+		LastUpdateDate    string   `xml:"LastUpdateDate"`
+		IsPublished       string   `xml:"IsPublished"`
+	}
+
+	ArticleExternal struct {
+		XMLName        xml.Name            `xml:"NewsArticleInformation"`
+		ClubName       string              `xml:"ClubName"`
+		ClubWebsiteURL string              `xml:"ClubWebsiteURL"`
+		NewsArticle    NewsArticleExternal `xml:"NewsArticle"`
+	}
+
+	NewsArticleExternal struct {
+		XMLName           xml.Name `xml:"NewsArticle"`
+		ArticleURL        string   `xml:"ArticleURL"`
+		NewsArticleID     int64    `xml:"NewsArticleID"`
+		PublishDate       string   `xml:"PublishDate"`
+		Taxonomies        string   `xml:"Taxonomies"`
+		TeaserText        string   `xml:"TeaserText"`
+		Subtitle          string   `xml:"Subtitle"`
+		ThumbnailImageURL string   `xml:"ThumbnailImageURL"`
+		Title             string   `xml:"Title"`
+		BodyText          string   `xml:"BodyText"`
+		GalleryImageURLs  string   `xml:"GalleryImageURLs"`
+		VideoURL          string   `xml:"VideoURL"`
+		OptaMatchId       string   `xml:"OptaMatchId"`
+		LastUpdateDate    string   `xml:"LastUpdateDate"`
+		IsPublished       string   `xml:"IsPublished"`
+	}
+)
+
+type (
+	NewsletterNewsItemDB struct {
```

Comment:
DB types are supposed to be defined on a repository level, no? otherwise you mix types from different layers

Reply:
fixed